PCA_Comp_Analysis_Dredging.R

```diff
@@ -510,30 +636,111 @@ write.csv(df, "C:/Users/ksanf/Documents/SD/DFs/dry_PC4.csv", row.names=FALSE)
+           air2m + pr_wtr+ shum925 +shum700, 
+         df, family=gaussian, na.action = "na.fail")
+
+# #dry season pc1
+# res<-glm(y~omega + merid_transport700 + slp + merid_trans925 + pottmp1000_500 + zonal_trans_925 + hgt500 + zonal_trans_700  + pottmp1000_850 +air2m + shum925 + shum700, df, family=gaussian, na.action = "na.fail")
+# #dry season pc2
+# res<-glm(y~omega + merid_transport700 +hgt1000 + airdiff + merid_trans925 + zonal_trans_925 + hgt500 + zonal_trans_700  + pottmp1000_850 + skt + pr_wtr + shum925 + shum700, df, family=gaussian, na.action = "na.fail")
+# #dry season pc3
+# res<-glm(y~omega + merid_transport700 + slp + airdiff + merid_trans925 + zonal_trans_925 + hgt500 + zonal_trans_700 + pottmp1000_850 + air2m + pr_wtr + shum925 + shum700, df, family=gaussian, na.action = "na.fail")
+# #dry season pc4
+# res<-glm(y~omega + merid_transport700 + hgt1000 + airdiff + merid_trans925 + zonal_trans_925 + hgt500 + zonal_trans_700  + pottmp1000_850 + air2m  + pr_wtr + shum925 + shum700, df, family=gaussian, na.action = "na.fail")
+
+# #dry season pc1
+res<-glm(y~omega + merid_transport700 + slp 
+         + merid_trans925 + pottmp1000_500 
+         + zonal_trans_925 + hgt500 
+         + zonal_trans_700  + pottmp1000_850 
+         +air2m + shum925 + shum700, 
+         df, family=gaussian, na.action = "na.fail")
+# #dry season pc2
+res<-glm(y~omega + merid_transport700 
+         +hgt1000 
+         #+ airdiff 
+         + merid_trans925 + zonal_trans_925 
+         + hgt500 + zonal_trans_700  + pottmp1000_850 
+         + skt + pr_wtr + shum925 + shum700, 
+         df, family=gaussian, na.action = "na.fail")
+# #dry season pc3
+res<-glm(y~omega + merid_transport700 + slp 
+         #+ airdiff 
+         + merid_trans925 + zonal_trans_925 + hgt500 
+         + zonal_trans_700 + pottmp1000_850 + air2m 
+         + pr_wtr + shum925 + shum700, 
+         df, family=gaussian, na.action = "na.fail")
+# #dry season pc4
+res<-glm(y~omega + merid_transport700 + hgt1000 
+         #+ airdiff 
+         + merid_trans925 + zonal_trans_925 
+         + hgt500 + zonal_trans_700  + pottmp1000_850 
+         + air2m  + pr_wtr + shum925 + shum700, 
+         df, family=gaussian, na.action = "na.fail")
+#</OET>
 
 
 ####################################################################################
 #all possible models
 dredge=dredge(res, rank = "AICc",extra="R^2")
 #save the dredge table to directory each time (change season name and pc number)
-write.csv(dredge,"C:/Users/ksanf/Documents/SD/Dredge_Tables/dry_PC4_dredge.csv", row.names = FALSE)
+filename=paste(season,"_PC",pcamode,"_dredge.csv",sep="")
+print("write predictand and predictor data table:")
+# if sub-folder does not exist create
+dhelp=file.path(outputdir.train,'dredge')
+if (!dir.exists(dhelp)){
+  dir.create(dhelp)
+  print("created subfolder:")
+  print(dhelp)
+}
+print(file.path(dhelp,filename))
+# OET write just the top ten models to output
+# 
+dredge_ret=10

```

Comment:
this code retains the first 10 models? i think in the next code, the dredge tables are reduced by certain criteria (e.g. AIC). Should we implement this here instead of the next code? Just in case more than 10 models were to be needed?

Reply:
I updated it in the devel branch: _ipass=dredge$delta<=delta_AIC_max
write.csv(dredge[ipass,],file.path(dhelp,filename), row.names = FALSE)_
where delta_AIC_max is now part of the user's control variables (set to 2 currently). So output is truncated to the significant models with same skill.